packages/components/src/composite/legacy/index.tsx

```diff
@@ -145,19 +150,16 @@ function proxyComposite< C extends Component >(
 // provided role, and returning the appropriate component.
 const unproxiedCompositeGroup = forwardRef<
 	any,
-	React.ComponentPropsWithoutRef<
-		typeof Current.CompositeGroup | typeof Current.CompositeRow
-	>
+	React.ComponentPropsWithoutRef< typeof Current.Group | typeof Current.Row >
 >( ( { role, ...props }, ref ) => {
-	const Component =
-		role === 'row' ? Current.CompositeRow : Current.CompositeGroup;
+	const Component = role === 'row' ? Current.Row : Current.Group;
 	return <Component ref={ ref } role={ role } { ...props } />;
 } );
 unproxiedCompositeGroup.displayName = 'CompositeGroup';
```

Comment:
I wanted to fix the issue in the legacy story where `CustomItem` had a display name of `Custom.Item`. I removed this line because I thought it was unnecessary, and somehow everything is fixed 😂

```suggestion
```

Reply:
Weird, maybe related to ?

Done in https://github.com/WordPress/gutenberg/pull/63564/commits/e4b29b2981052b0cd567d9f1926bd31bb81422ed